File: ToDoWeb/src/app/models/chore.ts
import { State } from "./state";

export interface ChoreData {
    statusCode: number;
    success: boolean;
    messages: string[];
    data: Chore[];
  }

export interface Chore {
    id: number;
    description: string;
    dueDate: string;
    state: State;
  }

  export interface CreateChore {
    description: string;
    dueDate: string;
    state: number;
  }

File: ToDoAPI/API/Data/Models/Chore.cs
using System;
using System.Collections.Generic;

namespace API.Data.Models;

public partial class Chore
{
    public int Id { get; set; }

    public string Description { get; set; } = null!;

    public DateTime DueDate { get; set; }

    public int StateId { get; set; }

    public virtual State State { get; set; } = null!;
}
